src/vs/editor/contrib/stickyScroll/browser/stickyScrollController.ts

```diff
@@ -3,63 +3,281 @@
 	private readonly _sessionStore: DisposableStore = new DisposableStore();
 
 	private _widgetState: StickyScrollWidgetState;
 	private _maxStickyLines: number = Number.MAX_SAFE_INTEGER;
 
+	private _stickyElements: HTMLCollection | undefined;
+	private _numberStickyElements: number | undefined;
+
+	private _stickyRangeProjectedOnEditor: IRange | undefined;
+	private _candidateDefinitionsLength: number = -1;
+
+	private _focused: boolean = false;
+	private _focusDisposableStore: DisposableStore | undefined;
+	private _focusedStickyElement: HTMLDivElement | undefined;
+	private _focusedStickyElementIndex: number | undefined;
+	private _stickyScrollFocusedContextKey: IContextKey<boolean>;
+
 	constructor(
 		private readonly _editor: ICodeEditor,
 		@IContextMenuService private readonly _contextMenuService: IContextMenuService,
-		@ILanguageFeaturesService languageFeaturesService: ILanguageFeaturesService,
-		@IInstantiationService instaService: IInstantiationService,
+		@ILanguageFeaturesService private readonly _languageFeaturesService: ILanguageFeaturesService,
+		@IInstantiationService private readonly _instaService: IInstantiationService,
+		@IContextKeyService private readonly _contextKeyService: IContextKeyService,
 	) {
 		super();
+		this._stickyScrollWidget = new StickyScrollWidget(this._editor);
+		this._stickyLineCandidateProvider = new StickyLineCandidateProvider(this._editor, this._languageFeaturesService);
+		this._register(this._stickyScrollWidget);
+		this._register(this._stickyLineCandidateProvider);
 
-		this._stickyScrollWidget = new StickyScrollWidget(this._editor, languageFeaturesService, instaService);
-		this._stickyLineCandidateProvider = new StickyLineCandidateProvider(this._editor, languageFeaturesService);
+		// The line numbers to stick
 		this._widgetState = new StickyScrollWidgetState([], 0);
 
-		this._register(this._stickyScrollWidget);
-		this._register(this._stickyLineCandidateProvider);
+		// Read the initial configuration to check if sticky scroll is enabled
+		this._readConfiguration();
+		// If the editor option for sticky scroll has changed, read the configuration again
 		this._register(this._editor.onDidChangeConfiguration(e => {
 			if (e.hasChanged(EditorOption.stickyScroll)) {
 				this._readConfiguration();
 			}
 		}));
-		this._readConfiguration();
+
 		this._register(dom.addDisposableListener(this._stickyScrollWidget.getDomNode(), dom.EventType.CONTEXT_MENU, async (event: MouseEvent) => {
 			this._onContextMenu(event);
 		}));
+
+		// Context key which indicates if the sticky scroll is focused or not
+		this._stickyScrollFocusedContextKey = EditorContextKeys.stickyScrollFocused.bindTo(this._contextKeyService);
+		// Create a focus tracker to track the focus on the sticky scroll widget
+		const focusTracker = this._register(dom.trackFocus(this._stickyScrollWidget.getDomNode()));
+		this._register(focusTracker.onDidBlur(_ => {
+			this._disposeFocusStickyScrollStore();
```

Comment:
Make sure to also listen on `onDidFocus` because your widget can receive dom focus through many ways, e.g tabbing into the widget will focus it. Something like: 

```
focusTracker.onDidFocus(_ => this.focus())
```